src/morx.rs

```diff
@@ -0,0 +1,1825 @@
+                        {
+                            liga_subst.next_state = 0;
+                            liga_subst.component_stack.clear();
+                            liga_subst.process_glyphs(ligature_subtable)?;
+                        }
+                    }
+                    4 => {
+                        if let SubtableType::NonContextual {
+                            noncontextual_subtable,
+                        } = &subtable.subtable_body
+                        {
+                            noncontextual_substitution(glyphs, noncontextual_subtable)?;
+                        }
+                    }
+                    _ => {}
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+//------------------------------------ Process requested feature list -----------------------------------------------------
+pub fn subfeatureflags(chain: &Chain, features: &Features) -> Result<u32, ParseError> {
+    //Feature type:
+    const LIGATURE_TYPE: u16 = 1;
+    //Feature selectors:
+    const COMMON_LIGATURES_ON: u16 = 2;
+    const COMMON_LIGATURES_OFF: u16 = 3;
+    const CONTEXTUAL_LIGATURES_ON: u16 = 18;
+    const CONTEXTUAL_LIGATURES_OFF: u16 = 19;
+    const HISTORICAL_LIGATURES_ON: u16 = 20;
+    const HISTORICAL_LIGATURES_OFF: u16 = 21;
+    //--------------------------------------
+    //Feature type:
+    const NUMBER_CASE_TYPE: u16 = 21;
+    //Feature selectors:
+    const OLD_STYLE_NUMBERS: u16 = 0;
+    const LINING_NUMBERS: u16 = 1;
+    //---------------------------------------
+    //Feature type:
+    const NUMBER_SPACING_TYPE: u16 = 6;
+    //Feature selectors:
+    const TABULAR_NUMBERS: u16 = 0;
+    const PROPORTIONAL_NUMBERS: u16 = 1;
+    //---------------------------------------
+    //Feature type:
+    const FRACTION_TYPE: u16 = 11;
+    //Feature selectors:
+    const NO_FRACTIONS: u16 = 0;
+    const FRACTIONS_STACKED: u16 = 1;
+    const FRACTIONS_DIAGONAL: u16 = 2;
+    //---------------------------------------
+    //Feature type:
+    const VERTICAL_POSITION_TYPE: u16 = 10;
+    //Feature selectors:
+    const ORDINALS: u16 = 3;
+    //---------------------------------------
+    //Feature type:
+    const TYPOGRAPHIC_EXTRAS_TYPE: u16 = 14;
+    //Feature selectors:
+    const SLASHED_ZERO_ON: u16 = 4;
+    const SLASHED_ZERO_OFF: u16 = 5;
+    //---------------------------------------
+    //Feature type:
+    const LOWERCASE_TYPE: u16 = 37;
+    //Feature selectors:
+    const LOWERCASE_SMALL_CAPS: u16 = 1;
+    //---------------------------------------
+    //Feature type:
+    const UPPERCASE_TYPE: u16 = 38;
+    //Feature selectors:
+    const UPPERCASE_SMALL_CAPS: u16 = 1;
+    //---------------------------------------
+
+    let mut subfeature_flags = chain.chain_header.default_flags;
+
+    for entry in chain.feature_array.iter() {
+        match features {
+            Features::Custom(_features_list) => {
+                return Ok(subfeature_flags);
```

Comment:
Should this return the supplied custom features instead of the default features?